### PR TITLE
FIO Request List themed.

### PR DESCRIPTION
**Pull request by andreyvEze**
Pixel 3
Снимок экрана 2021-01-13 в 23 56 40

### CODE REVIEW

**Comment by paullinator** [comment]
@andreyvEze please make the theming PR‘s against the UI2 branch not develop

File: src/components/scenes/FioRequestListScene.js [hunk 1]
```diff
@@ -542,45 +534,45 @@ export class FioRequestList extends React.Component<Props, LocalState> {
...
       <SceneWrapper background="header">
         {fullScreenLoader && <FullScreenLoader indicatorStyles={styles.fullScreenLoader} />}
         <View style={styles.scene}>
           <View style={styles.row}>
-            <SettingsHeaderRow icon={<Image source={fioRequestsIcon} style={styles.iconImage} />} text={s.strings.fio_pending_requests} />
+            <ListHeader title={s.strings.fio_pending_requests} />
             {!loadingPending && !fioRequestsPending.length ? (
               <View style={styles.emptyListContainer}>
-                <T style={styles.text}>{s.strings.fio_no_requests_label}</T>
+                <EdgeText style={styles.text}>{s.strings.fio_no_requests_label}</EdgeText>
```
Comment (andreyvEze) [inline src/components/scenes/FioRequestListScene.js:549]: remove `emptyListContainer`  , use only  EdgeText component

File: src/components/scenes/FioRequestListScene.js [hunk 2]
```diff
@@ -542,45 +534,45 @@ export class FioRequestList extends React.Component<Props, LocalState> {
...
               />
             </View>
           </View>
           <View style={styles.row}>
-            <SettingsHeaderRow icon={<IonIcon name="ios-send" color={THEME.COLORS.WHITE} size={this.headerIconSize} />} text={s.strings.fio_sent_requests} />
+            <ListHeader title={s.strings.fio_sent_requests} withTopMargin />
             {!loadingSent && !fioRequestsSent.length ? (
               <View style={styles.emptyListContainer}>
-                <T style={styles.text}>{s.strings.fio_no_requests_label}</T>
+                <EdgeText style={styles.text}>{s.strings.fio_no_requests_label}</EdgeText>
```
Comment (andreyvEze) [inline src/components/scenes/FioRequestListScene.js:570]: remove `emptyListContainer` , use only EdgeText component

File: src/components/themed/ListHeader.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  title: string,
+  withTopMargin?: boolean
+}
+
+class ListHeaderComponent extends React.PureComponent<Props & ThemeProps> {
+  render() {
+    const styles = getStyles(this.props.theme)
+    return (
+      <View style={[styles.container, this.props.withTopMargin ? styles.topMargin : null]}>
+        <EdgeText style={styles.title}>{this.props.title}</EdgeText>
```
Comment (andreyvEze) [inline src/components/themed/ListHeader.js:19]: add option for `children`
rename component for `UnderlinedHeader`

File: src/components/scenes/FioRequestListScene.js [hunk 4]
```diff
@@ -622,48 +607,65 @@ const rawStyles = {
...
-const styles: typeof rawStyles = StyleSheet.create(rawStyles)
+}))
+
+const FioRequestListScene = connect(
+  (state: RootState) => {
+    const fioWallets = getFioWallets(state)
+    const wallets = getWallets(state)
+    const wallet = getSelectedWallet(state)
+    const account = state.core.account
+    if (!wallet) {
```
Comment (andreyvEze) [inline src/components/scenes/FioRequestListScene.js:643]: remove this case, add checks in component

File: src/modules/FioRequest/components/FioRequestRow.js [hunk 5]
```diff
@@ -63,99 +58,143 @@ class FioRequestRow extends React.Component<Props> {
...
-            <View style={styles.transactionRight}>
-              <View style={[styles.transactionDetailsRow, fioRequest.content.memo ? styles.transactionDetailsRowMargin : null]}>
-                <T style={styles.transactionPartner} adjustsFontSizeToFit minimumFontScale={this.minimumFontScale}>
-                  {isSent ? fioRequest.payer_fio_address : fioRequest.payee_fio_address}
-                </T>
-                {this.currencyField(fioRequest.content.amount, isSent ? fioRequest.status : '')}
+            <View style={styles.requestRight}>
+              <View style={styles.requestDetailsRow}>
+                <EdgeText style={styles.name}>{isSent ? fioRequest.payer_fio_address : fioRequest.payee_fio_address}</EdgeText>
+                {this.currencyField(fioRequest.content.amount)}
```
Comment (andreyvEze) [inline src/modules/FioRequest/components/FioRequestRow.js:119]: remove method and put logic in the `render`

File: src/modules/FioRequest/components/FioRequestRow.js [hunk 6]
```diff
@@ -63,99 +58,143 @@ class FioRequestRow extends React.Component<Props> {
...
               </View>
-              <View style={styles.transactionDetailsRow}>
+              <View style={styles.requestDetailsRow}>
                 {this.requestedTimeAndMemo(new Date(fioRequest.time_stamp), fioRequest.content.memo)}
```
Comment (andreyvEze) [inline src/modules/FioRequest/components/FioRequestRow.js:122]: remove method and put logic in the `render`

File: src/modules/FioRequest/components/FioRequestRow.js [hunk 7]
```diff
@@ -63,99 +58,143 @@ class FioRequestRow extends React.Component<Props> {
...
-          onPress={this.onSelect}
-          underlayColor={this.underlayColor}
-          style={[styles.singleTransaction, { borderBottomWidth: isLastOfDate ? 0 : 1 }]}
-        >
-          <View style={styles.transactionInfoWrap}>
-            <View style={styles.transactionLeft}>
-              <View style={styles.transactionLeftLogoWrap}>
-                <Image style={styles.transactionLogo} source={fioRequestsIcon} />
-              </View>
+      <TouchableHighlight onPress={this.onSelect} underlayColor={theme.backgroundGradientLeft}>
```
Comment (andreyvEze) [inline src/modules/FioRequest/components/FioRequestRow.js:109]: create clickable row component

File: src/components/scenes/FioRequestListScene.js [hunk 8]
```diff
@@ -486,28 +485,21 @@ export class FioRequestList extends React.Component<Props, LocalState> {
...
-    const isLastOfDate =
-      index + 1 === this.state.fioRequestsSent.length ||
-      (index > 0 && formatDate(new Date(this.state.fioRequestsSent[index + 1].time_stamp)) !== formatDate(new Date(fioRequest.time_stamp)))
-    return <FioRequestRow fioRequest={fioRequest} onSelect={this.selectSentRequest} isSent isLastOfDate={isLastOfDate} />
+  renderSent = (itemObj: { item: FioRequest }) => {
+    const { item: fioRequest } = itemObj
+    return <FioRequestRow fioRequest={fioRequest} onSelect={this.selectSentRequest} isSent />
   }
 
   renderHiddenItem = (rowObj: { item: FioRequest }, rowMap: { [string]: SwipeRow }) => {
```
Comment (andreyvEze) [inline src/components/scenes/FioRequestListScene.js:505]: HiddenMenuButtons should have the same behavior as current wallet list swipe row

**Pull request closed without merging**